### PR TITLE
Add K8S service sample

### PR DESCRIPTION
**Pull request by baixinsui**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,100 +60,105 @@ billing:
...
+
+      variable "k8s_node_count" {
+      type = number
+      description = "The count of nodes except master of the K8S cluster."
+      }
+
+      resource "huaweicloud_compute_instance" "k8s-master" {
+      name = "k8s-master"
+      availability_zone = data.huaweicloud_availability_zones.osc-az.names[0]
+      flavor_id = "c7.xlarge.4"
```
Comment (niuzhenguo) [inline README.md:122]: better to make this c7.xlarge.4 configurable, mybe with different flavors type like performance、normal...

Comment (baixinsui) [reply README.md:122]: Ok. I will define a variable flavor_id and use it here

File: README.md [hunk 2]
```diff
@@ -60,100 +60,105 @@ billing:
...
-      - name: secgroup_id
-        description: The secgroup id.
-        kind: variable
-        type: string
-        mandatory: false
-        validator:
-      - name: number_test
-        description: Validator range for value of dataType number.
+      - name: k8s_node_count
+        description: The node except master count of K8S cluster.
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply README.md:90]: Ok.

File: README.md [hunk 3]
```diff
@@ -29,13 +29,13 @@ interacting with the fundamental APIs:
 
 ```yaml
 # The version of the OCL
-version: 2.0
+version: 1.0
 # The category of the service.
 category: middleware
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply README.md:34]: Ok. I will Change the catagory to "container". Is it ok?

**Review by iskey — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -82,78 +82,84 @@ deployment:
...
+      count = var.worker_nodes_count
+      name = "k8s-node-${count.index}"
+      availability_zone = data.huaweicloud_availability_zones.osc-az.names[0]
+      flavor_id = var.node_flavor_id
+      network {
+          uuid = "6a7c406d-5bc6-4b50-8f75-948aea69cb24"
+      }
+
+      security_group_ids = [ "668833c8-459d-4d15-871d-34c3cbcda9d4" ]
+      image_id = "ef88c527-b556-4199-bae5-3e80fba1733e"
```
Comment (iskey) [inline README.md:146]: avoid to use the `image_id` directly, we can use data resources to filter the right image with some special tags.

File: README.md [hunk 5]
```diff
@@ -82,78 +82,84 @@ deployment:
...
+      type = string
+      description = "The flavor id of all nodes in the K8S cluster."
+      }
+
+      resource "huaweicloud_compute_instance" "k8s-master" {
+      name = "k8s-master"
+      availability_zone = data.huaweicloud_availability_zones.osc-az.names[0]
+      flavor_id = var.node_flavor_id
+      network {
+          uuid = "6a7c406d-5bc6-4b50-8f75-948aea69cb24"
```
Comment (iskey) [inline README.md:125]: We can settle the network here to be an optional context variable. we can give the user a chance to choose the proper id, or use the default network if nothing provided.

File: README.md [hunk 6]
```diff
@@ -82,78 +82,84 @@ deployment:
...
+      }
+      security_group_ids = [ "668833c8-459d-4d15-871d-34c3cbcda9d4" ]
```
Comment (iskey) [inline README.md:127]: The same as the network.

File: samples/K8S-cluster.yaml [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+# The Service provided by the ISV, the name will be shown on the console as a service.
+name: K8S cluster
+# The version of the service, if the end-user want to select the version when they want to deploy the service.
+serviceVersion: v1.26.2
+# For the users may have more than one service, the @namespace can be used to separate the clusters.
+description: This is an ehanced kafka services by ISV-A.
+namespace: ISV-A
+# Icon for the service.
+icon: |
+  data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAACRAQMAAAAPc4+9AAAAAXNSR0IB2cksfwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAZQTFRF+/v7Hh8gVD0A0wAAAcVJREFUeJzNlc1twzAMhSX44KNH0CgaTd6gK3kUd4McDVTwq/hjiUyaIk
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply samples/K8S-cluster.yaml:14]: Ok. I will use the  k8s icon from https://kubernetes.io/

File: samples/K8S-cluster.yaml [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+      master: 1
+deployment:
+  # kind, Supported values are terraform, pulumi, crossplane.
+  kind: terraform
+  # Context for deployment: the context including some kind of parameters for the deployment, such as fix,variable.
+  # - env: The value of the fix parameters are defined by the ISV with the @value at the initial time.
+  # - variable: The value of the variable parameters are defined by the user on the console.
+  # The parameters will be used to generate the API of the managed service.
+  context:
+    - name: worker_nodes_count
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply samples/K8S-cluster.yaml:55]: Ok. I will resume these env variables

File: samples/K8S-cluster.yaml [hunk 9]
```diff
@@ -0,0 +1,133 @@
+# The version of the OCL
+version: 2.0
+# The category of the service.
+category: container
+# The Service provided by the ISV, the name will be shown on the console as a service.
+name: K8S cluster
+# The version of the service, if the end-user want to select the version when they want to deploy the service.
+serviceVersion: v1.26.2
+# For the users may have more than one service, the @namespace can be used to separate the clusters.
+description: This is an ehanced kafka services by ISV-A.
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply samples/K8S-cluster.yaml:10]: Ok.

File: README.md (hunk 1 above)
Comment (iskey) [reply README.md]: no `flavor_id` please, the deployment already be able to pass the flavor name to the deploy process yet.

File: README.md (hunk 5 above)
Comment (baixinsui) [reply README.md:125]: Ok. I will add a variable 'network_id' with a default value.

File: README.md (hunk 6 above)
Comment (baixinsui) [reply README.md:127]: Ok. I will add a variable 'security_group_id' with a default value.

File: README.md (hunk 1 above)
Comment (baixinsui) [reply README.md]: I define the performance and normal flavor in flavors. The property of flavor contains worker_nodes_count, flavor_id, master_nodes_count.

File: README.md (hunk 4 above)
Comment (baixinsui) [reply README.md:146]: Ok. I use data image to find image. Now this image can found in project 'osc' and 'default'

**Pull request closed without merging**